File: 计算机基础/算法案例/1.两数之和.js
/*
 * @lc app=leetcode.cn id=1 lang=javascript
 *
 * [1] 两数之和
 */

/** 
 * 给定一个整数数组 nums 和一个目标值 target，请你在该数组中找出和为目标值的那 两个 整数，并返回他们的数组下标。
 * 你可以假设每种输入只会对应一个答案。但是，数组中同一个元素不能使用两遍。 
 */ 

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  let map = {}
  for (let i = 0; i < nums.length; i++) {
    let temp = target - nums[i]
    if (typeof map[temp] === 'number') {
      return [map[temp], i]
    }
    map[nums[i]] = i
  }
  return [-1, -1]
};

twoSum([2, 7, 11, 15], 9)
// @lc code=end

File: LeetCode/1.两数之和.py
'''
Author: li.haoran li.haoran@99cloud.net
Date: 2022-05-31 18:59:42
LastEditors: li.haoran li.haoran@99cloud.net
LastEditTime: 2022-05-31 19:15:29
'''
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start
class Solution:
  def twoSum(self, nums, target: int):
    hashtable = dict()
    for i, num in enumerate(nums):
      if target - num in hashtable:
        return [hashtable[target - num], i]
      hashtable[nums[i]] = i
    return []

    # @lc code=end
